scene/main/node.cpp

```diff
@@ -1149,9 +1148,8 @@ void Node::set_name(const String &p_name) {
 	data.name = name;
 
 	if (data.parent) {
-		data.parent->data.children.erase(old_name);
 		data.parent->_validate_child_name(this, true);
-		data.parent->data.children.insert(data.name, this);
+		ERR_FAIL_COND(!data.parent->data.children.replace_key(old_name, data.name));
```

Comment:
I think the preferred code style here is to make a mutating call like that be explicitely separated:
```suggestion
		bool success = data.parent->data.children.replace_key(old_name, data.name);
		ERR_FAIL_COND(!success);
```
Example from the same file:
https://github.com/godotengine/godot/blob/afc5fa14adce7676746662751e63eab2d2ad4420/scene/main/node.cpp#L1439-L1440